numerics/elementary_functions_body.hpp

```diff
@@ -117,6 +117,62 @@ constexpr Q NextDown(Q const& x) {
   return si::Unit<Q> * numerics::_next::NextDown(x / si::Unit<Q>);
 }
 
+template<int exponent>
+constexpr double Pow(double x) {
+  // Use the Russian peasant algorithm for small exponents.
+  if constexpr (exponent > 0 && exponent < 32) {
+    // The end of the recursion is handled by the specializations below.
+    auto const y = Pow<exponent / 2>(x);
+    auto const y² = y * y;
+    if constexpr (exponent % 2 == 1) {
+      return y² * x;
+    } else {
+      return y²;
+    }
+  } else if constexpr (exponent < 0 && exponent > -32) {
+    return 1 / Pow<-exponent>(x);
+  } else {
+    return std::pow(x, exponent);
+  }
+}
+
+// Static specializations for frequently-used exponents, so that this gets
+// turned into multiplications at compile time.
+
+template<>
+inline constexpr double Pow<0>(double) {
+  return 1;
+}
+
+template<>
+inline constexpr double Pow<1>(double x) {
+  return x;
+}
+
+template<>
+inline constexpr double Pow<2>(double x) {
+  return x * x;
+}
+
+template<>
+inline constexpr double Pow<3>(double x) {
+  return x * x * x;
+}
+
+template<int exponent, typename Q>
+constexpr Exponentiation<Q, exponent> Pow(Q const& x) {
```

Comment:
FWIW, this function is horrible and I was trying to clean it up in #4272.  A discussion for another day.